### ISSUE TITLE
Ordered comparisons

### ISSUE THREAD

**Issue by Armavica** [astral-sh/ruff#9459]
The rule `SIM300` discourages so-called "Yoda comparisons", i.e. comparisons where the numerical constant is first. While this can make sense to focus on the variable, this can lead to code that is more difficult to read and more easy to get wrong, especially for interval checking. See for example https://llewellynfalco.blogspot.com/2016/02/dont-use-greater-than-sign-in.html for the full argument.

Consider the following code:

``` python
def f(x):
    a = (5 < x and x < 10)  # noqa: SIM300 # probably the easiest to read though
    b = (5 < x and 10 > x)  # noqa: SIM300
    c = (x > 5 and x < 10)
    d = (x > 5 and 10 > x)  # noqa: SIM300
    e = (x < 10 and 5 < x)  # noqa: SIM300
    f = (x < 10 and x > 5)
    g = (10 < x and 5 < x)  # noqa: SIM300
    h = (10 < x and x < 5)  # noqa: SIM300
    return (a, b, c, d, e, f, g, h)
```

Out of the eight different ways to write this condition, I would argue that the first one is by far the one whose intent is the clearest, despite the first condition being flagged by SIM300.

I could see several possible options for a rule or rules fixing this pattern:
* Don't do anything: this does not address the pattern, but also does not contradict SIM300
* Add a rule "Don't use > or >=" for simple comparisons, contradicting SIM300 in some cases
* Address the specific case of range checking (like in this example) by additionally reordering the left and right boundaries, still contradicting SIM300 in some cases

**Comment by tjkuson** [comment]
The argument linked seems to be that the author finds one style easier to read and extrapolates that other developers shouldn't use the greater sign for consistency (it could just as well be titled 'Don't use the less than sign in programming' or 'Always put the numerical constant on the right').

Personally, I find that keeping the variable on the left makes comparisons significantly easier to read and like that `SIM300` enforces this (I prefer `x > 5` to `5 < x`), so would prefer that Ruff keeps _some way_ of linting for this.

**Comment by Armavica** [comment]
Yes, it is a convention and as with many others some might prefer the opposite one. Perhaps rule SIM300 could be configurable, to either flag one or the other pattern?

**Comment by hauntsaninja** [comment]
Maybe the best thing here is if the fix recognises the situation and suggests the [chained comparison](https://docs.python.org/3/library/stdtypes.html#comparisons) `5 < x < 10` instead.
